mlflow/utils/rest_utils.py

```diff
@@ -38,6 +38,7 @@
 _REST_API_PATH_PREFIX = "/api/2.0"
 _UC_OSS_REST_API_PATH_PREFIX = "/api/2.1"
 _TRACE_REST_API_PATH_PREFIX = f"{_REST_API_PATH_PREFIX}/mlflow/traces"
+_ARMERIA_OK = "200 OK"
```

Comment:
This was a kludge I suggested.  If you have a better suggestion here @dbczumar or @BenWilson2, happy to change this.  The problem is that the armeria server that UC OSS us using seems to require that an OK be sent back with the 200 which breaks the current parse_dict.

Reply:
I'd probably put this logic within the validation function that handles errors and request payload inconsistencies like this one.

```python
def verify_rest_response(response, endpoint):
    """Verify the return code and format, raise exception if the request was not successful."""
    
    # Handle Armeria-specific response case where response text is "200 OK"
    if response.status_code == 200 and response.text.strip() == _ARMERIA_OK:
        response._content = b'{}'  # Update response content to be an empty JSON dictionary
        return response

    # Handle non-200 status codes
    if response.status_code != 200:
        if _can_parse_as_json_object(response.text):
            raise RestException(json.loads(response.text))
        else:
            base_msg = (
                f"API request to endpoint {endpoint} "
                f"failed with error code {response.status_code} != 200"
            )
            raise MlflowException(
                f"{base_msg}. Response body: '{response.text}'",
                error_code=get_error_code(response.status_code),
            )

    # Skip validation for endpoints that may return a non-JSON response
    if endpoint.startswith(_REST_API_PATH_PREFIX) and not _can_parse_as_json_object(response.text):
        base_msg = (
            "API request to endpoint was successful but the response body was not "
            "in a valid JSON format"
        )
        raise MlflowException(f"{base_msg}. Response body: '{response.text}'")
    
    return response
```

Reply:
completed